doc/modules/clustering.rst

```diff
@@ -1582,3 +1582,35 @@ Drawbacks
  *  Caliński, T., & Harabasz, J. (1974). "A dendrite method for cluster
     analysis". Communications in Statistics-theory and Methods 3: 1-27.
     `doi:10.1080/03610926.2011.560741 <http://dx.doi.org/10.1080/03610926.2011.560741>`_.
+
+.. _contingency_matrix:
+
+Contingency Matrix
+----------------------
+
+Contingency matrix is a format describing the relationship between
+discrete-valued random variables.
+It shows the multivariate frequency distribution of the variables.
+
+Let the first random variable be :math:`X`, and the second random
+variable be :math:`Y`.
+In a contingency matrix :math:`M`, entry :math:`M(i, j)` is the number of
+observations with :math:`X=X_i` and :math:`Y=Y_j`.
+
+Here is an example::
+
+   >>> from sklearn.metrics.cluster import contingency_matrix
+   >>> x = ["male", "male", "male", "female", "female", "female"]
+   >>> y = ["right_handed", "right_handed", "left_handed", "left_handed",
+   ...      "ambidextrous", "ambidextrous"]
+   >>> contingency_matrix(x, y)
+   array([[2, 1, 0],
+          [0, 1, 2]])
+
+Confusion matrix is a special case of contingency matrix with
+two dimensions.
```

Comment:
Contingency matrix and confusion matrix are both 2d. Confusion matrix is square and the order of columns and rows is significant.